pkgs/os-specific/linux/kernel/perf/default.nix

```diff
@@ -63,7 +63,7 @@ stdenv.mkDerivation {
 
   # Fix 6.10.0 holding pkg-config completely wrong.
   # Patches from perf-tools-next, should be in 6.11 or hopefully backported.
-  patches = lib.optionals (lib.versionAtLeast kernel.version "6.10") [
+  patches = lib.optionals (lib.versions.majorMinor kernel.version == "6.10") [
```

Comment:
unrelated, but is there any reason we need to build perf for every kernel version?

Reply:
Not really AFAIK.

Reply:
You sure that this fixes perf? Fails for me @K900 @alyssais 

```
rror: builder for '/nix/store/7m0fhi4mdib79x91fadkq17d0xwvrmjs-perf-linux-6.11.drv' failed with exit code 1;
       last 17 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/xrhpizkr37hc6zj55ab2spsnl3gr8h64-linux-6.11.tar.xz
       > source root is linux-6.11
       > setting SOURCE_DATE_EPOCH to timestamp 1726412276 of file linux-6.11/virt/lib/irqbypass.c
       > Running phase: patchPhase
       > applying patch /nix/store/cklsdshnrdvfifqvh9v1wn403n18yds6-?id=0f0e1f44569061e3dc590cd0b8cb74d8fd53706b
       > patching file tools/build/feature/test-libtracefs.c
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 1 out of 1 hunk ignored -- saving rejects to file tools/build/feature/test-libtracefs.c.rej
       > patching file tools/perf/Makefile.config
       > Hunk #1 FAILED at 183.
       > Hunk #2 FAILED at 1178.
       > Hunk #3 FAILED at 1194.
       > Hunk #4 FAILED at 1315.
       > 4 out of 4 hunks FAILED -- saving rejects to file tools/perf/Makefile.config.rej
```

Reply:
Yes it does, I have it built right now on my system.

Reply:
Hm. Any idea why it fails on my side?

I'm using `packages = [ config.boot.kernelPackages.perf ]`

Reply:
Interesting. My failure only happens with `kernelPackages_6_11` from `nixos-24.05`